tests/test_devcli.py

```diff
@@ -269,3 +265,248 @@ def test_new_plugin_path_and_name_params(project_cli_runner):
     path = 'path'
     assert set(os.listdir(path)) == set(
         ['lektor_plugin_name.py', 'setup.py', '.gitignore', 'README.md'])
+
+
+# new-theme
+def test_new_theme(project_cli_runner):
+    result = project_cli_runner.invoke(
+        cli, ["dev", "new-theme"],
+        input='Lektor Theme Name\n'
+        '\n'
+        'Author Name\n'
+        'author@email.com\n'
+        'y\n',
+    )
+    assert "Create Theme?" in result.output
+    assert result.exit_code == 0
+    path = os.path.join('themes', 'lektor-theme-theme-name')
+    assert set(os.listdir(path)) == set(
+        ['example-site', 'images', 'README.md', 'theme.ini'])
+    assert set(os.listdir(os.path.join(path, 'images'))) == set(
+        ['homepage.png'])
+    assert set(os.listdir(os.path.join(path, 'example-site'))) == set(
+        ['lektor-theme-theme-name.lektorproject', 'README.md', 'themes'])
+    try:
+        assert os.readlink(os.path.join(
+            path, 'example-site/themes/lektor-theme-theme-name')) == \
+            '../../../lektor-theme-theme-name'
+    except AttributeError:
+        pass
+
+    # theme.ini
+    theme_ini_expected = textwrap.dedent("""
+        [theme]
+        description =
+        features =
+        homepage =
+        lektor_required_version = 3.1
+        license = MIT
+        licenselink =
+        name = Lektor Theme Name
+        tags =
+
+        [author]
+        email = author@email.com
+        name = Author Name
+        homepage =
+
+        # If porting an existing theme
+        [original]
+        author =
+        homepage =
+        repo =
+    """).strip()
+    with open(os.path.join(path, 'theme.ini')) as f:
+        theme_ini_contents = f.read().strip()
+    assert theme_ini_contents == theme_ini_expected
```

Comment:
This is a matter of taste I guess, but isn't it better to test the parsed output? That seems closer to reality and what we guarantee. Reordering fields or adding a new field or section is not a breaking change.

Reply:
Yes, you're right. I've changed it to only test for the values we construct, reading it as the ini it is. So then there's a general test of structure, and then the few specific things that are more likely to break.